pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -188,44 +216,105 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
 			}
 			return reconcile.Result{}, nil
 		}
-		var pbgetErr error
-		pb, pbgetErr = r.getProfileBundleFromRulesOrVars(instance)
-		if pbgetErr != nil && !common.IsRetriable(pbgetErr) {
-			// the Profile or ProfileBundle objects didn't exist. Surface the error.
-			err = r.handleTailoredProfileStatusError(instance, pbgetErr)
-			return reconcile.Result{}, err
-		} else if pbgetErr != nil {
-			return reconcile.Result{}, pbgetErr
+
+		// we will not use ProfileBundle for CustomRules
+		if len(customRules) <= 0 {
+			reqLogger.Info("No Custom Rules found for TailoredProfile", "tailoredProfile", instance.GetName())
+			var pbgetErr error
+			pb, pbgetErr = r.getProfileBundleFromRulesOrVars(instance)
+			if pbgetErr != nil && !common.IsRetriable(pbgetErr) {
+				// the Profile or ProfileBundle objects didn't exist. Surface the error.
+				err = r.handleTailoredProfileStatusError(instance, pbgetErr)
+				return reconcile.Result{}, err
+			} else if pbgetErr != nil {
+				return reconcile.Result{}, pbgetErr
+			}
+			// Make TailoredProfile be owned by the ProfileBundle. This way
+			// we can ensure garbage collection happens.
+			// This update will trigger a requeue with the new object.
+			// We will skip this if a customRule is being used.
+			if needsControllerRef(instance) {
+				tpCopy := instance.DeepCopy()
+				anns := tpCopy.GetAnnotations()
+				if anns == nil {
+					anns = make(map[string]string)
+				}
+				// If the user already provided the product type, we
+				// don't need to set it
+				_, ok := anns[cmpv1alpha1.ProductTypeAnnotation]
+				if !ok {
+					if strings.HasSuffix(tpCopy.GetName(), "-node") {
+						anns[cmpv1alpha1.ProductTypeAnnotation] = string(cmpv1alpha1.ScanTypeNode)
+					} else {
+						anns[cmpv1alpha1.ProductTypeAnnotation] = string(cmpv1alpha1.ScanTypePlatform)
+					}
+					tpCopy.SetAnnotations(anns)
+				}
+				// This will trigger an update anyway
+				return r.setOwnership(tpCopy, pb)
+			}
 		}
 
-		// Make TailoredProfile be owned by the ProfileBundle. This way
-		// we can ensure garbage collection happens.
-		// This update will trigger a requeue with the new object.
-		if needsControllerRef(instance) {
-			tpCopy := instance.DeepCopy()
-			anns := tpCopy.GetAnnotations()
-			if anns == nil {
-				anns = make(map[string]string)
+		if len(customRules) > 0 {
+			reqLogger.Info("Custom rules detected, skipping ProfileBundle and setting product type to platform")
+			// we are detection custom rules, we will not use ProfileBundle, and at the moment we only support platform type
```

Comment:
This ultimately means TailoredProfiles with CustomRules won't get cleaned up when users delete ProfileBundles, since the CustomRules are not linked to ProfileBundles in anyway.